src/components/dialogs/network-modifications/voltage-level-topology-modification/voltage-level-topology-modification-form.tsx

```diff
@@ -0,0 +1,290 @@
+                ...row,
+                [CURRENT_CONNECTION_STATUS]:
+                    row[CURRENT_CONNECTION_STATUS] === undefined ? null : row[CURRENT_CONNECTION_STATUS],
+            }));
+            replace(updatedData);
+        }
+    }, [rowData, replace]);
+
+    const defaultColDef = {
+        sortable: false,
+        resizable: true,
+        wrapHeaderText: true,
+        editable: false,
+        headerClass: 'centered-header',
+    };
+
+    const columnDefs: ColumnDef[] = useMemo(
+        () => [
+            {
+                field: SWITCH_ID,
+                filter: true,
+                flex: 1,
+                cellStyle: (params: CellStyleParams) => {
+                    return params.data.isSeparator
+                        ? {
+                              backgroundColor: '#f0f0f0',
+                              fontWeight: 'bold',
+                          }
+                        : {};
+                },
+                headerComponent: HeaderComponentWithTooltip,
+                headerComponentParams: {
+                    displayName: intl.formatMessage({ id: 'switchId' }),
+                    tooltipTitle: intl.formatMessage({
+                        id: isNodeBuilt(currentNode) ? 'builtNodeTooltipVlTopoModif' : 'notBuiltNodeTooltipVlTopoModif',
+                    }),
+                    isNodeBuilt: isNodeBuilt(currentNode),
+                    disabledTooltip: !isUpdate && isNodeBuilt(currentNode),
+                },
+            } as ColumnDef,
+            {
+                field: PREV_CONNECTION_STATUS,
+                flex: 1,
+                cellStyle: (params: CellStyleParams): Record<string, string> => {
+                    if (params.data.isSeparator) {
+                        return {
+                            backgroundColor: '#f0f0f0',
+                            fontWeight: 'bold',
+                        };
+                    } else {
+                        return {};
+                    }
+                },
+                cellClass: 'ag-cell-center',
+                headerComponent: HeaderComponentWithTooltip,
+                headerComponentParams: {
+                    displayName: intl.formatMessage({ id: 'previousStatus' }),
+                    tooltipTitle: intl.formatMessage({
+                        id: isNodeBuilt(currentNode) ? 'builtNodeTooltipVlTopoModif' : 'notBuiltNodeTooltipVlTopoModif',
+                    }),
+                    isNodeBuilt: isNodeBuilt(currentNode),
+                    disabledTooltip: !isUpdate && isNodeBuilt(currentNode),
+                },
+            } as ColumnDef,
+            {
+                field: CURRENT_CONNECTION_STATUS,
+                flex: 1,
+                cellRenderer: (params: CellRendererParams) => {
+                    if (params.data.isSeparator) {
+                        return '';
+                    }
+                    return BooleanNullableCellRenderer(params);
```

Comment:
we should determine whether this is a functional component or technical component to choose the right name and typing props
If a technical component, it is aimed to reuse for other cell render params, so need a small transformation
If a functional component, typing directly with cell render params